src/notifications/reducer.ts

```diff
@@ -0,0 +1,84 @@
+import { createReducer } from 'redux-act'
+
+import * as actions from './actions'
+
+export interface State {
+    readNotificationList: {
+        notifications: any[]
+        resultExhausted: boolean
+    }
+    unreadNotificationList: any[]
+    showMoreIndex?: any
+    unreadNotifications: number
+    currentPage: number
+    isLoading: boolean
+    isReadExpanded: boolean
+    showInbox: boolean
+    unreadNotifCount: number
+    shouldTrack: boolean
+}
+
+const defaultState: State = {
+    readNotificationList: {
+        notifications: [],
+        resultExhausted: false,
+    },
+    unreadNotificationList: [],
+    showMoreIndex: undefined,
+    unreadNotifications: 0,
+    currentPage: 0,
+    isLoading: true,
+    isReadExpanded: false,
+    showInbox: false,
+    unreadNotifCount: 0,
+    shouldTrack: true,
```

Comment:
For now `shouldTrack` is the reducer only for tracking, but for making it generic make it generic and the key of the checked switches will be there. That make sense?

Reply:
TBH I don't really understand. A few ways around this:
1. for N notifs with toggles, make N dynamic bool states that set some values under keys that correspond to the toggles (maybe each toggle could be assigned a unique `id` prop which you could also use as the key - would need to be passed to the reducer)
2. update `ToggleSwitch` to __optionally__ be a controlled input (if `props.isChecked` is not defined) and manage it's own internal state

I would recommend 2 as it would make `ToggleSwitch` more easily re-usable in different contexts and it should be pretty simple IMO, but you're the one implementing it